Source/missiles.cpp

```diff
@@ -3093,7 +3093,8 @@ void MI_Fireball(Missile &missile)
 			AddUnLight(missile._mlid);
 		}
 	} else {
-		int dam = missile._midam;
+		auto &monster = Monsters[id];
+		int dam = (missile._micaster == TARGET_MONSTERS) ? missile._midam : monster.mMinDamage + GenerateRnd(monster.mMaxDamage - monster.mMinDamage + 1);
 		MoveMissileAndCheckMissileCol(missile, dam, dam, true, false);

```

Comment:
```suggestion
		int minDam = missile._midam;
		int maxDam = missile._midam;
		
		if (missile._micaster != TARGET_MONSTERS) {
			minDam = monster.mMinDamage;
			maxDam = monster.mMaxDamage;
		}
		MoveMissileAndCheckMissileCol(missile, minDam, maxDam, true, false);
```

This lets the damage roll be applied in PlayerMHit, letting the duplicate code be removed from here.

Reply:
Hmm, apparently it reuses the same damage roll for the splash damage. I guess it shouldn't make much of a difference to just have it reroll damage in each tile.